src/cls/IPM/Lifecycle/Base.cls

```diff
@@ -1807,7 +1721,129 @@ Method GetResourceRelativePath(pResource As %String, pExtension As %String) As %
 	Quit tDirectory_tFile
 }
 
-ClassMethod GetBaseClassForPackaging(pPackaging As %String, Output pBaseClass As %Dictionary.Classname) As %Status [ Final ]
+Method ExportSingleModule(
+	ByRef pResourceArray,
+	ByRef pTargetDirectory As %String,
+	ByRef pDependencyGraph,
+	ByRef pParams,
+	pVerbose As %Boolean = 0)
+{
+	Merge tParams = pParams
+
+	Set tFullResourceName = ""
+	For {
+		Set tFullResourceName = $Order(pResourceArray(tFullResourceName),1,tSrcModule)
+		If (tFullResourceName = "") {
+			Quit
+		}
+		
+		If $Get(pResourceArray(tFullResourceName,"Generated")) Continue
+		
+		Set tName = $Piece(tFullResourceName,".",1,*-1)
+		Set tExt = $Piece(tFullResourceName,".",*)
+		If '$Data(pResourceArray(tFullResourceName,"RelativePath"),tRelativePath) {
+			Set tRelativePath = ..GetResourceRelativePath(tName,tExt)
+		}
+		
+		If (tExt = "ZPM") && (tName '= ..Module.Name) {
+			Set tRelativePath = pDependencyGraph(tName,"Location")_tRelativePath
+		} ElseIf (tSrcModule '= "") && (tSrcModule '= ..Module.Name) {
+			Set tRelativePath = pDependencyGraph(tSrcModule,"Location")_tRelativePath
+		}
+		
+		Set tFullPath = ##class(%File).NormalizeFilename(pTargetDirectory_tRelativePath)
+		Set tFullPath = ##class(%File).NormalizeFilenameWithSpaces(tFullPath)
+
+		// Handle FileCopy different --> set target directory from dependency graph
+		If ($Get(pResourceArray(tFullResourceName, "Processor")) '= "") && $Get(pResourceArray(tFullResourceName, "Processor")).%IsA("%IPM.ResourceProcessor.FileCopy") {
```

Comment:
Would such explicit hard coding for FileCopy affect other resource processors? For example, in HS, we have AngularArtifact and JavaLibrary. Ideally we should be consistent with what is passed to OnExportItem for the base module and dependencies so it can behave the same way but its inputs will guarantee that the files go to their correct locations (whether its for the base module or under .modules/ for dependencies)

Reply:
Agreed, we should just do this across the board.